numpyro/distributions/truncated.py

```diff
@@ -353,41 +365,10 @@ def TruncatedDistribution(base_dist, low=None, high=None, validate_args=None):
         )
 
 
-class TruncatedCauchy(LeftTruncatedDistribution):
-    arg_constraints = {
-        "low": constraints.real,
-        "loc": constraints.real,
-        "scale": constraints.positive,
-    }
-    reparametrized_params = ["low", "loc", "scale"]
-
-    def __init__(self, low=0.0, loc=0.0, scale=1.0, validate_args=None):
-        self.low, self.loc, self.scale = promote_shapes(low, loc, scale)
-        super().__init__(
-            Cauchy(self.loc, self.scale), low=self.low, validate_args=validate_args
-        )
-
-    @property
-    def mean(self):
-        return jnp.full(self.batch_shape, jnp.nan)
-
-    @property
-    def variance(self):
-        return jnp.full(self.batch_shape, jnp.nan)
-
-    def tree_flatten(self):
-        if isinstance(self._support.lower_bound, (int, float)):
-            aux_data = self._support.lower_bound
-        else:
-            aux_data = None
-        return (self.low, self.loc, self.scale), aux_data
-
-    @classmethod
-    def tree_unflatten(cls, aux_data, params):
-        d = cls(*params)
-        if aux_data is not None:
-            d._support = constraints.greater_than(aux_data)
-        return d
+def TruncatedCauchy(*, loc=0.0, scale=1.0, low=None, high=None, validate_args=None):
```

Comment:
I'm thinking that we might relax the number of keyword arguments a bit
```
# sample for TruncatedNormal
def TruncatedCauchy(loc=0.0, scale=1.0, *, low=None, high=None, validate_args=None):
```
Current users will still get enough information to fix their code if they use `(low, loc, scale)` arguments. New users will have a simpler way to specify their distributions, like `dist.TruncatedNormal(0, 1, high=10.)` What do you think?

Reply:
Sounds good to me!

If someone previous passed:
- `TruncatedNormal(0, 0, 1)`, they'll get an error
- `TruncatedNormal(0, loc=0, scale=1)` they'll get an error
- `TruncatedNormal(low=0, loc=0, scale=1)` their code will keep working

I don't see any downsides, and I agree that it would benefit new users, so I'll make the change :+1: